### PR TITLE
feat: week time unit

### LINKED ISSUE #6349: Add week Time Unit
Hi,
it could be very useful to be able to aggregate by week, or to have a generic step definition aggregation.

Thank you

### PR DESCRIPTION
**Pull request by domoritz**
Also remove time unit namespace, support more time units, improve docs

fixes #6349

### CODE REVIEW

**Comment by domoritz** [comment]
cc @kristw

**Comment by domoritz** [comment]
I think this may also improve performance since I moved a bunch of definitions to the tests.

File: site/docs/transform/timeunit.md [hunk 1]
```diff
@@ -8,17 +8,22 @@ Time unit is used to discretize times in Vega-Lite. It can be used (1) with the
...
+- `"milliseconds"` - Milliseconds in a second.
+
+Vega-Lite time units can also be a string of consecutive time units to indicate desired intervals of time. For example, `yearmonthdate` indicates chronological time sensitive to year, month, and date (but not to hours, minutes, or seconds). The specifier `monthdate` is sensitive to month and date, but not year, which can be useful for binning time values to look at seasonal patterns only.
+
+Vega-Lite currently supports the following intervals: `"yearquarter"`, `"yearquartermonth"`, `"yearmonth"`, `"yearmonthdate"`, `"yearmonthdatehours"`, `"yearmonthdatehoursminutes"`, `"yearmonthdatehoursminutesseconds"`, `"yearweek"`, `"yearweekday"`, `"yearweekdayhours"`, `"yearweekdayhoursminutes"`, `"yearweekdayhoursminutesseconds"`, `"quartermonth"`, `"monthdate"`, `"monthdatehours"`, `"monthdatehoursminutes"`, `"monthdatehoursminutesseconds"`, `"weekday"`, `"weeksdayhours"`, `"weekdayhoursminutes"`, `"weekdayhoursminutesseconds"`, `"dayhours"`, `"dayhoursminutes"`, `"dayhoursminutesseconds"`, `"hoursminutes"`, `"hoursminutesseconds"`, `"minutesseconds"`, `"secondsmilliseconds"`.
```
[the remark this answers is not in the document]
Comment (kanitw) [reply site/docs/transform/timeunit.md:24]: I also don't know if we need to list all of them here.  The more important point is to mention that users can combine them?  (Who would read this list anyway?
)

[the remark this answers is not in the document]
Comment (domoritz) [reply site/docs/transform/timeunit.md:24]: re intervals. I copied from https://vega.github.io/vega/docs/transforms/timeunit/. 

I can remove the list.

**Comment by domoritz** [comment]
Addressed comments and rebased.

**Pull request merged**